client/state/products-list/assembler.js

```diff
@@ -1,4 +1,11 @@
-export const createProductObject = ( product ) => {
-	product.cost = Number( product.cost );
-	return product;
-};
+/**
+ * Converts the `cost` of a product to a Number, if it isn't already.
+ *
+ * @param {object} product A product object.
+ * @returns A shallow copy of the original product, but with `cost` set to the
+ * 	 equivalent numeric value.
+ */
+export const ensureNumericCost = ( product ) => ( {
+	...product,
+	cost: Number( product.cost ),
```

Comment:
Are we fine with `NaN` or `Infinity` here? Should we make it `0` in those cases? 

For the protocol, if `product.cost` is `'test'`, it will end up being `NaN`. If we pass a huge number of just `'Infinity'`, it'll result in being `Infinity`. Those are valid `Number` values 😉 The question here is: are we handling those correctly where this cost is being displayed?

I'm also curious why is this necessary. Is it because prices arrive at strings from the API response?

Reply:
Honestly I couldn't tell you the reasoning behind the original logic. It looks like it originated in #30684. The API endpoint is set to always cast this field's value to a `float` before returning, so with the way PHP behaves I assume that guarantees we always get a floating-point number back, even if the original value is null or non-numeric to start with. Do you think it's safe to cut this function all together?

Reply:
I've verified that this will always come as a float from the REST API 👍 It's safe to keep it as-is, because the values I mentioned can't end up being in the `cost` field.

Let's keep the function just in case - there probably has been a reason to have it in the first place.